helm/cluster-azure/values.yaml

```diff
@@ -5,6 +5,7 @@ connectivity:
     instanceType: Standard_D2s_v5
   network:
     controlPlane:
+      apiServerLbIp: 10.0.0.2
```

Comment:
i've tried to create a private cluster from this branch and only set `network.mode=private` but cluster-creation stuck with

```
I0322 11:33:25.412712       1 recorder.go:103] events "msg"="failed to reconcile cluster services: failed to reconcile AzureCluster service loadbalancers: failed to create resource mariopriv02/mariopriv02-api-internal-lb (servi
ce: loadbalancers): network.LoadBalancersClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code=\"PrivateIPAddressInReservedRange\" Message=\"Private static IP address 10.0.0.2 falls within reser
ved IP range of subnet prefix 10.0.0.0/20.\" Details=[]" "object"={"kind":"AzureCluster","namespace":"org-multi-project","name":"mariopriv02","uid":"c425df82-8caa-41d1-915b-6ea3c838fce7","apiVersion":"infrastructure.cluster.x-k
8s.io/v1beta1","resourceVersion":"92316264"} "reason"="ClusterReconcilerNormalFailed" "type"="Warning"  
```

Reply:
I don't think we should have a default value here since that depends on the subnet used .

should we actually auto generate it from the subnet value ?

Reply:
> i've tried to create a private cluster from this branch and only set connectivity.network.mode="private"

Setting only `connectivity.network.mode="private"` cannot work, as we also have to customize VNet network range in order to deploy a workload cluster to a network range that is accessible from our VPN and that is also not overlapping with any other WC from that MC.

> According to the error message, the apiServerLbIp must be outside any nodeCIDR range, or?

It has to be from the control plane subnet CIDR.

We should generate not only this value, but also all CIDRs as well (VNet and all subnets), so they don't overlap with MC or with other WCs.

We already have a story for [validating if network ranges are valid and non-overlapping](https://github.com/giantswarm/roadmap/issues/1985), so you immediately get an error if your config is not valid. This validation would be the first improvement.

We should also have another story for automatically assigning available network ranges (e.g. in kubectl-gs, or defaulting in admission controllers, or something else), so that we don't have to specify them manually (e.g. in vintage aws-operator and azure-operator are doing this). These are some of the reasons why I didn't want to make private clusters a default option in this pull requests.

As for this specific `apiServerLbIp` field, since it should come from the configured control plane subnet CIDR, auto-generating it in Helm based on the specified control plane subnet CIDR wouldn't be very pretty, also not sure how reliable. For now I'll try to do some string/int manipulation and see how it looks/works, but we should have this implemented in some place other than Helm (see the paragraph above) and keep the templates as simple as possible.

Reply:
Thanks for your explanation.

I would also prefer a "validation/generation" logic outside helm (as helm is not the right tool for doing this - IMHO).

Reply:
And almost forgot one of the most important bits for `apiServerLbIp` :smile: In addition to the need of specifying network ranges, an IP address `x.x.x.2` is really invalid address (for any kind of use case), because `x.x.x.0` - `x.x.x.3` are reserved IPs in Azure subnets and cannot be used :facepalm: Not sure how I didn't catch the error before, as I also tested creation of public clusters from my branch where I think I tried the default values.